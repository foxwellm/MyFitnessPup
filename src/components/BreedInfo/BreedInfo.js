import React from 'react'
import AboutBreeds from '../AboutBreeds/AboutBreeds';
import { Link } from 'react-router-dom'

export const BreedInfo = (props) => {

  return (
    <div>
      <AboutBreeds />
      <div className='breed-info-container'>
        <div className='breed-info-card'>
          <div className='back-btn-container'>
            <Link to='/about-breeds'>
              <button className='breed-info-btn'>Back</button>
            </Link>
          </div>
          <p>{props.info.desc}</p>
          <div className='breed-info-img'>
            <img className='breed-img' alt='dog' src={require(`../../assets/images/${props.breed.img}`)} />
          </div>
        </div>
      </div>
    </div>
  )
}

export default BreedInfo