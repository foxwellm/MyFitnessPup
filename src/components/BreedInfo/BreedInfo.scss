.breed-info-container {
  background-color: rgba(0, 0, 0, 0.7);
  left: 0;
  top: 0;
  height: 100%;
  width: 100%;
  position: fixed;
} 

.breed-info-card {
  background-color: $primary-light;
  font-family: $font-secondary;
  border: 1px solid black;
  padding: 20px;
  border-radius: 5px;
  left: 20vw;
  top: 20vh; 
  box-shadow: 2px 2px 6px black;
  width: 60vw;
  height: 60vh;
  min-height: 300px;
  min-width: 600px;
  position: fixed;
}

.breed-info-btn {
  text-decoration: none;
  cursor: pointer;
  transition: .1s;
  text-align: center;
  border: 0;
  font-size: 15px;
  color: #fcfcfc;
  background-color: $primary-dark;
  height: 35px;
  width: 80px;
  border-radius: 5px;
  outline: none;
  &:hover {
    transform: scale(1.01);
    background-color: $primary-dark-hover;
    transition: .1s;
  }
}

.back-btn-container {
display: flex;
justify-content: flex-start;
}

.breed-info-img {
  position: relative;
  width: 40vw;
  height: 20vw;
  margin: auto;
  background-size: 40vw 20vw;
}

.breed-img {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
}