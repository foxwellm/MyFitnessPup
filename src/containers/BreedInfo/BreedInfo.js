import React, {Component} from 'react'
import AboutBreeds from '../../components/AboutBreeds/AboutBreeds';
import {Link} from 'react-router-dom'
import {importImages} from '../../helpers/importImages'

export class BreedInfo extends Component {

  render() {
    const {img} = this.props.breed
    const images = importImages();
    return (
      <div>
        <AboutBreeds />
        <div className='breed-info-container'>
        <div className='breed-info-card'>
            <div className='back-btn-container'><Link to='/about-breeds'><button className='breed-info-btn'>Back</button></Link></div>
            <p>{this.props.info.desc}</p>
            <div className='breed-info-img' style={{ backgroundImage: `url(${images[img]})` }}></div>
        </div>
        </div>
      </div>
    )
  }
}

export default BreedInfo