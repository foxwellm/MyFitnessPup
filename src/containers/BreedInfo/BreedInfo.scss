.breed-info-container {
  background-color: rgba(0, 0, 0, 0.7);
  left: 0;
  top: 0;
  height: 100%;
  width: 100%;
  position: fixed;
} 

.form {
  background-color: $primary-light;
  font-family: $font-secondary;
  border: 1px solid black;
  padding: 20px;
  border-radius: 5px;
  // margin-left: 50%;
  // margin-top: 50%;
  left: 20vw;
  top: 20vh; 
  box-shadow: 2px 2px 6px black;
  width: 60vw;
  height: 60vh;
  position: fixed;
  // z-index: 1;
  // display: flex;
  // flex-direction: column;
  // max-height: 450px;
}