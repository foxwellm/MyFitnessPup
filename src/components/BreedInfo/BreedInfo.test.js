import React from 'react'
import { shallow } from 'enzyme'
import BreedInfo from './BreedInfo'

describe('BreedInfo', () => {
  let wrapper
  let mockBreed
  let mockInfo
  it('should have match the correct snapshot', () => {
    mockBreed = {
      breed: 'Siberian Husky',
      tag: 'siberian-husky',
      img: 'siberian-husky.jpg',
      isCold: true,
      isRunner: true,
      isClimber: true
    }
    mockInfo = {
      name: 'Siberian Husky', desc: "The Siberian Husky was originally bred to pull sleds and carts long distances in the harsh Russian climate.They are powerfully built dogs with lively spirits, always ready for adventure at any time.A true endurance breed, Siberians make an excellent choice for those looking for a dog that will keep them entertained on long hikes.They are gentle and alert, but very social dogs that enjoy spending time outside."
    }

    wrapper = shallow(<BreedInfo breed={mockBreed} info={mockInfo} />)
    expect(wrapper).toMatchSnapshot()
  })
})