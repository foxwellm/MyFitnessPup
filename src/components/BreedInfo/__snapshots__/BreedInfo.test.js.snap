// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedInfo should have match the correct snapshot 1`] = `
<div>
  <AboutBreeds />
  <div
    className="breed-info-container"
  >
    <div
      className="form"
    />
  </div>
</div>
`;
